scenes/menu/preferences_window.tscn

```diff
@@ -0,0 +1,113 @@
+[gd_scene load_steps=2 format=3 uid="uid://dx5u3h8ppeab0"]
+
+[ext_resource type="Script" path="res://scenes/menu/preferences_window.gd" id="1_ttn5r"]
+
+[node name="PreferencesWindow" type="PopupPanel"]
+title = "Settings"
+initial_position = 1
+size = Vector2i(468, 485)
+visible = true
+borderless = false
+keep_title_visible = true
+script = ExtResource("1_ttn5r")
+
+[node name="ColorRect" type="ColorRect" parent="."]
+offset_left = 4.0
+offset_top = 4.0
+offset_right = 464.0
+offset_bottom = 481.0
+color = Color(4.21144e-07, 0.300865, 0.300865, 1)
+
+[node name="VBoxContainer" type="VBoxContainer" parent="."]
+offset_left = 4.0
+offset_top = 4.0
+offset_right = 464.0
+offset_bottom = 481.0
+
+[node name="BGMCheckBox" type="CheckBox" parent="VBoxContainer"]
+layout_mode = 2
+theme_override_font_sizes/font_size = 30
+button_pressed = true
+text = "Main Menu Background Music"
+
+[node name="GameSoundsCheckBox" type="CheckBox" parent="VBoxContainer"]
+layout_mode = 2
+theme_override_font_sizes/font_size = 30
+button_pressed = true
+text = "Game Turn Sound"
+
+[node name="HSeparator" type="HSeparator" parent="VBoxContainer"]
+layout_mode = 2
+
+[node name="RoomSettingsLabel" type="Label" parent="VBoxContainer"]
+layout_mode = 2
+theme_override_font_sizes/font_size = 32
+text = "Custom Room Creation"
+
+[node name="BestOfLabel" type="Label" parent="VBoxContainer"]
+layout_mode = 2
+theme_override_font_sizes/font_size = 28
+text = "Best Of:"
+
+[node name="BestOfSelection" type="OptionButton" parent="VBoxContainer"]
+layout_mode = 2
+disabled = true
+item_count = 5
+selected = 0
+popup/item_0/text = "1"
+popup/item_0/id = 1
+popup/item_1/text = "3"
+popup/item_1/id = 3
+popup/item_2/text = "5"
+popup/item_2/id = 5
+popup/item_3/text = "7"
+popup/item_3/id = 7
+popup/item_4/text = "∞"
+popup/item_4/id = 0
+
+[node name="StartingTimersLabel" type="Label" parent="VBoxContainer"]
+layout_mode = 2
+theme_override_font_sizes/font_size = 28
+text = "Starting Timers:"
+
+[node name="StartingTimersSelection" type="OptionButton" parent="VBoxContainer"]
+layout_mode = 2
+item_count = 7
+selected = 3
+popup/item_0/text = "6:00"
```

Comment:
Only in intervals of 3?
I would just put everything from 5 to 20. I doubt anyone is going to play an enforced 20+ minute game because why?

Reply:
For the per move overtime setting, let's have 10, 15, 20, 25, 30 as options.

Reply:
if eventually serious tournaments start being played on the client it seems very likely to me that a 30 minute option would be the standard, and I think having that option would be appealing to the type of people who consider the pinnacle of tournament play to be unwatchable 2.5 hour tts best of 5s. to be honest adding a 30 minute option and wanting a 12 minute option for the casual tournament are the reasons I wanted it to be possible to fiddle with the clock settings in the first place

Reply:
30 minutes each is so long!
I guess I don't really care, but can I ask you to make the options added programmatically so we can easily change it?
That will also make it easier to add more options/change them as desired.

Reply:
now programmatic